src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl

```diff
@@ -98,10 +98,11 @@ idle(cast, {csvList,Workers,CSVData}, State = #source_statem_state{chunkSize = C
 %%This cast spawns a transmitter of data stream towards NerlClient by casting batches of data from parsed csv file given by cowboy source_server
 idle(cast, {startCasting,Body}, State = #source_statem_state{myName = MyName, lengthOfSample = LengthOfSample, sendingMethod = Method, frequency = Frequency, chunkSize = ChunkSize, sourcePid = [],workersMap = WorkersMap, castingTo = CastingTo, nerlnetGraph = NerlnetGraph, msgCounter = Counter, csvName = CSVName, csvList =CSVlist}) ->
     [_Source,NumOfBatchesToSend] = re:split(binary_to_list(Body), ",", [{return, list}]),
+    NumBatches = list_to_integer(NumOfBatchesToSend),
 
   io:format("start casting to: ~p~n, number of batches to send: ~p~ntotal casting list length: ~p~n ",[CastingTo,NumOfBatchesToSend, length(CSVlist)]),
-
-  Transmitter =  spawnTransmitter(CastingTo,CSVName,CSVlist,NerlnetGraph,MyName,WorkersMap,ChunkSize,LengthOfSample,Frequency,list_to_integer(NumOfBatchesToSend),Method) ,
+  NumToSend = if NumBatches < length(CSVlist) -> NumBatches; true -> length(CSVlist) end,
```

Comment:
What is CSVlist? 
How are num of batches and csvlist related?
Please add some documentation at this part to explain these questions.

Reply:
added documentation